### ISSUE TITLE
localstack IAM list operations (role and user) do not respect PathPrefix 

### ISSUE THREAD

**Issue by rpattcorner** [localstack/localstack#2747]
# Type of request: This is a ...

[ X] bug report
[ ] feature request

# Detailed description
The IAM list operations [list_users](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/iam.html#IAM.Client.list_users) and [list_roles](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/iam.html#IAM.Client.list_roles) have an optional PathPrefix attribute that guarantees retrieval of users and roles with a specified PathPrefix

Both PathPrefix qualifiers work in actual boto3 calls but fail with localstack endpoints

## Expected behavior
Only users or roles with the specified Path equal to the PathPrefix are listed

## Actual behavior
All users or roles are listed regardless of Path or PathPrefix.  The problem is symmetrical, i.e. a list specifying a defaulted path incorrectly returns items with a specified path, and a list specifying a path returns items with a defaulted path

## Code to illustrate the problem
```
#  Show problem localstack does not honor path
import localstack_client.session

session = localstack_client.session.Session()
iam = session.client('iam')
assume_role_policy_doc = '{"Version": "2012-10-17", "Statement": [{"Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::123412341234:root"]}, "Action": "sts:AssumeRole"}]}'

# clear out roles and users for a clean test
try:
    response_role = iam.delete_role(RoleName='fooRole')
    response_user = iam.delete_user(UserName='fooUser')
except Exception as e:
    pass

# Create a unique role with a clearly defined path
response = iam.create_role(
    Path='/somePathOrOther/',
    RoleName='fooRole',
    AssumeRolePolicyDocument='{"Version": "2012-10-17", "Statement": [{"Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::123412341234:root"]}, "Action": "sts:AssumeRole"}]}',
    Description='string'
)

#  Look for the role where it *should not be found*, in the default path
#  The path format follows the aws rules of starting and ending with a slash, but a naked path gives same result
role_list = iam.list_roles(PathPrefix='/')["Roles"]
if len(role_list) != 0:
    for role in role_list:
        print ("We asked for a list of roles with a PathPrefix of '/' but found the role with name \'{}\' and path \'{}\'".
               format(role["RoleName"], role["Path"]))

#  Users have the same problem
response = iam.create_user(
    Path='/somePathOrOther/',
    UserName='fooUser',
)

#  Look for the role where it *should not be found*, in the default path
user_list = iam.list_users(PathPrefix='/')["Users"]
if len(role_list) != 0:
    for user in user_list:
        print ("We asked for a list of users with a PathPrefix of '/' but found the user with name \'{}\' and path \'{}\'".
               format(user["UserName"], role["Path"]))

#  Generally the localstack implementation of list_users and list_roles symmetrically ignores the path, e.g.
#  creating an entity with a default (/) path shows up when you ask for a more specific path

#  Clear out again
try:
    response_role = iam.delete_role(RoleName='fooRole')
    response_user = iam.delete_user(UserName='fooUser')
except Exception as e:
    pass

response = iam.create_role(
    RoleName='fooNoPathRole',   # Note, path is missing, reverts to slash
    AssumeRolePolicyDocument='{"Version": "2012-10-17", "Statement": [{"Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::123412341234:root"]}, "Action": "sts:AssumeRole"}]}',
    Description='string'
)

#  Look for the default path role where it *should not be found*, in a specific path
role_list = iam.list_roles(PathPrefix='/somePathOrOther/')["Roles"]
if len(role_list) != 0:
    for role in role_list:
        print ("We asked for a list of roles with a PathPrefix of '/somePathOrOther/' but found the role with name \'{}\' and path \'{}\'".
               format(role["RoleName"], role["Path"]))
```

## Here's the result of running the code above:
```
We asked for a list of roles with a PathPrefix of '/' but found the role with name 'fooRole' and path '/somePathOrOther/'
We asked for a list of users with a PathPrefix of '/' but found the user with name 'fooUser' and path '/somePathOrOther/'
We asked for a list of roles with a PathPrefix of '/somePathOrOther/' but found the role with name 'fooNoPathRole' and path '/'

Process finished with exit code 0
```

## Command used to start LocalStack
docker-compose up with a default configuration
...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

...

**Comment by usmangani1** [comment]
@ninhkd This looks like a duplicate of the issue #2767 which is already closed.

**Comment by whummer** [comment]
Thanks for reporting @rpattcorner, this should be fixed via #2771 and spulec/moto#3180 - can you please do a `docker pull localstack/localstack` and give it another try with the `latest` version of the Docker image? Please let us know if the problem persists. Thanks!
